### PR TITLE
[PFSP1-0001][PFSP1-0002] Initial Screen, Login Screen

### LINKED ISSUE #20: [PFSP1-0001] Initial Screen
### Priority 
- [x] Low
- [ ] Mid
- [ ] High

### Links
Name | Link
-|-
User Story | [TBD]()
Figma | [link](https://www.figma.com/file/YWJimHMl9fq3v4dQBZSuQo/Hi-Fi-User?node-id=269%3A108)

### Completion
Expected | Time
-|-
Expected Start | 27-07-2020
Expected End | 27-07-2020

### LINKED ISSUE #21: [PFSP1-0002] Login Screen
### Priority 
- [ ] Low
- [ ] Mid
- [x] High

### Links
Name | Link
-|-
User Story | [TBD]()
Figma | [link](https://www.figma.com/file/YWJimHMl9fq3v4dQBZSuQo/Hi-Fi-User?node-id=71%3A91)

### Completion
Expected | Time
-|-
Expected Start | 28-07-2020
Expected End | 29-07-2020

### PR DESCRIPTION
**Pull request by evan03121998**
Closes #20
Closes #21 
***
## Description
Initial Screen View

## Related Links
List related PRs against other branches:

List | link
------ | ------
User Story | [link]()
Figma | [link]()
Issue | [link]()

## Todos
- [X] UI
- [ ] Coordinator
- [ ] Network
- [ ] Tests
- [ ] Documentation

## Screenshot / Video
Screenshot consisting of changes made on this branch.

### UI Changes
UI related changes will be shown here (can be either screenshot or video format)

### Network Status
Network related changes will be shown here (screenshot of request status)

### Test Coverage
Test coverage result for designated changes will be shown here (screenshot of coverage)

## Notes
Additional notes that you have either technical debt, blocker, etc.

### CODE REVIEW

File: MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        // Pass the selected object to the new view controller.
+    }
+    */
```
Comment (inxudianz) [inline MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift]: comment2 dihapus bang

File: MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+class InitialScreenViewController: UIViewController {
+
+    @IBOutlet weak var initialScreenImageView: UIImageView!
+    @IBOutlet weak var initialScreenLabel: UILabel!
+    @IBOutlet weak var initialScreenArrowButton: UIButton!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
```
Comment (inxudianz) [inline MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift]: commend dihapus

File: MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
+import UIKit
+
+class InitialScreenViewController: UIViewController {
+
+    @IBOutlet weak var initialScreenImageView: UIImageView!
+    @IBOutlet weak var initialScreenLabel: UILabel!
+    @IBOutlet weak var initialScreenArrowButton: UIButton!
```
Comment (inxudianz) [inline MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift:15]: buat IBAction untuk navigate to next screen sama buat coordinator

**Review by inxudianz — CHANGES_REQUESTED** [review]

File: MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift (hunk 1 above)
Comment (evan03121998) [reply MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift]: g

File: MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift (hunk 2 above)
Comment (evan03121998) [reply MainApp/MainApp/Feature/Splash/InitialScreenViewController.swift]: g

**Review by robbyaww — DISMISSED** [review]

**Review by inxudianz — CHANGES_REQUESTED** [review]
Space - space yang berlebihan dihapus

Coordinator related file taro di Coordinator/Feature group

**Review by inxudianz — CHANGES_REQUESTED** [review]
Login coordinator ga ke commit

File: MainApp/MainApp/Feature/Login/LoginViewController.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    @IBOutlet weak var facebookLoginButton: UIButton!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+    }
+
+    @IBAction func loginTapped(_ sender: Any) {
+        viewModel?.gotoHome()
```
Comment (inxudianz) [inline MainApp/MainApp/Feature/Login/LoginViewController.swift:29]: logic login email sama gitu2 nya di handle juga ya, tutorialnya entar besok sore

**Review by inxudianz — CHANGES_REQUESTED** [review]

**Review by robbyaww — DISMISSED** [review]

File: MainApp/MainApp/FeatureNetwork/SampleNetwork/LoginNetwork.swift [hunk 5]
```diff
@@ -0,0 +1,45 @@
+//
+//  LoginNetwork.swift
+//  MainApp
+//
+//  Created by Evan Christian on 06/08/20.
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
```
Comment (inxudianz) [inline MainApp/MainApp/FeatureNetwork/SampleNetwork/LoginNetwork.swift]: Pindahin LoginNetwork di group LoginNetwork

File: MainApp/MainApp/FeatureNetwork/SampleNetwork/LoginNetwork.swift [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import Foundation
+import RxSwift
```
Comment (inxudianz) [inline MainApp/MainApp/FeatureNetwork/SampleNetwork/LoginNetwork.swift]: RXSwift no need to be imported

File: MainApp/MainApp/Network/Service/LoginService.swift [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        switch self {
+        case .loginRequest:
+            return .post
+        }
+    }
+    
+    var task: HTTPTask {
+        switch self {
+        case .loginRequest:
+            return .plainRequest
```
Comment (inxudianz) [inline MainApp/MainApp/Network/Service/LoginService.swift]: login karena ada pake model pakenya parameteredRequest

File: MainApp/MainApp/Network/Path/FeaturePath/Sample/LoginPath.swift [hunk 8]
```diff
@@ -0,0 +1,13 @@
+//
+//  LoginPath.swift
+//  MainApp
+//
+//  Created by Evan Christian on 06/08/20.
```
Comment (inxudianz) [inline MainApp/MainApp/Network/Path/FeaturePath/Sample/LoginPath.swift]: pindahin ke FeaturePath/Auth/<File>

File: MainApp/MainApp/Network/Path/FeaturePath/Sample/LoginSubPath.swift [hunk 9]
```diff
@@ -0,0 +1,13 @@
+//
+//  LoginSubPath.swift
+//  MainApp
+//
+//  Created by Evan Christian on 06/08/20.
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
```
Comment (inxudianz) [inline MainApp/MainApp/Network/Path/FeaturePath/Sample/LoginSubPath.swift]: pindahin ke FeaturePath/Auth/<File>

**Review by inxudianz — CHANGES_REQUESTED** [review]

**Review by inxudianz — APPROVED** [review]

**Pull request merged**